table/tables/tables.go

```diff
@@ -1024,3 +1027,118 @@ func init() {
 	table.TableFromMeta = TableFromMeta
 	table.MockTableFromMeta = MockTableFromMeta
 }
+
+// ctxForPartitionExpr implement sessionctx.Context interfact.
+type ctxForPartitionExpr struct {
+	sessionVars *variable.SessionVars
+}
+
+// newCtxForPartitionExpr creates a new sessionctx.Context.
+func newCtxForPartitionExpr() sessionctx.Context {
+	sctx := &ctxForPartitionExpr{
+		sessionVars: variable.NewSessionVars(),
+	}
+	sctx.sessionVars.MaxChunkSize = 2
+	sctx.sessionVars.StmtCtx.TimeZone = time.UTC
+	return sctx
+}
+
+// NewTxn creates a new transaction for further execution.
+// If old transaction is valid, it is committed first.
+// It's used in BEGIN statement and DDL statements to commit old transaction.
+func (ctx *ctxForPartitionExpr) NewTxn() error {
+	panic("not support")
+}
+
+// Txn returns the current transaction which is created before executing a statement.
+func (ctx *ctxForPartitionExpr) Txn() kv.Transaction {
+	panic("not support")
+}
+
+// GetClient gets a kv.Client.
+func (ctx *ctxForPartitionExpr) GetClient() kv.Client {
+	panic("not support")
+}
+
+// SetValue saves a value associated with this context for key.
+func (ctx *ctxForPartitionExpr) SetValue(key fmt.Stringer, value interface{}) {
+	panic("not support")
+}
+
+// Value returns the value associated with this context for key.
+func (ctx *ctxForPartitionExpr) Value(key fmt.Stringer) interface{} {
+	panic("not support")
+}
+
+// ClearValue clears the value associated with this context for key.
+func (ctx *ctxForPartitionExpr) ClearValue(key fmt.Stringer) {
+	panic("not support")
+}
+
+func (ctx *ctxForPartitionExpr) GetSessionVars() *variable.SessionVars {
+	return ctx.sessionVars
```

Comment:
why not panic here as other functions?

Reply:
`expression.simpleRewriter` may use it if `BuildCastFunction` is called.